### PR TITLE
[rtl] refactor decoder to use rvdecoderdb

### PR DESCRIPTION
**Pull request by Lucas-Wye**
- remove the operations of reading inst-table.adoc
- using API from rvdecoderdb

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -86,7 +108,9 @@ trait T1
   def arithmeticModule = arithmetic
   def tilelinkModule = tilelink
   def hardfloatModule = hardfloat
+  def rvdecoderdbModule = rvdecoderdb
 
+  def cdeModule = cde
   def chiselModule = Some(chisel)
```
Comment (sequencer) [inline build.sc:114]: No cde in t1

File: t1/src/decoder/Decoder.scala [hunk 2]
```diff
@@ -628,8 +802,10 @@ object Decoder {
       sWrite,
       //sRead1 -> vType
       vtype,
+
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:805]: Nit don’t add space here

File: t1/src/decoder/Decoder.scala [hunk 3]
```diff
@@ -31,7 +33,182 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
...
+          // csr instructions
+          case s if Seq("vsetivli", "vsetvli", "vsetvl").contains(s) => false
+          // conflict instrctions `vmv` and `vmerge`, as defined in `riscv-v-spec/inst-table.adoc`
+          case s if s.contains("vmv.v") => false
+          case _ => true
+      }
+    }.toSeq.distinct
+
+
+    val expandedOps: Array[Op] = (instructions.filter(_.encoding.toString.substring(32-6-1, 32-0) == opVOpcode).map{ insn =>
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:176]: Kinda dirty... but OK...

File: t1/src/decoder/Decoder.scala [hunk 4]
```diff
@@ -31,7 +33,182 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
...
+        val funct3 = if (op2vFunct3.contains(realFunct3)) "V" else if (op2xFunct3.contains(realFunct3)) "X" else if (op2iFunct3.contains(realFunct3)) "I" else if (op2fFunct3.contains(realFunct3)) "F" else ""
+        val funct6 = insn.encoding.toString.substring(32-31-1, 32-26)
+        val special = insnSpec.collectFirst { case s if (insn.name.contains(s.instrName)) => SpecialAux(s.name, s.vs, s.value) }
+        val name = special match {
+          // remove postfix `vv`, `vx`, `vs`, `vi`, `mm`, `wi`, `v.v`, `v.i`, `v.x` from non-special instructions
+          case None => insn.name.replace(".vv", "").replace(".vx", "").replace(".vs", "").replace(".vi", "").replace(".mm", "").replace(".wi", "").replace(".v.v", "").replace(".v.i", "").replace(".v.x", "")
+            // replace `wx` and `wv` for instructions class `vwaddu.w` , `vfwadd.w` , `vwadd.w` , `vwsubu.w` , `vfwsub.w` , `vwsub.w`
+            .replace(".wx", ".w").replace(".wv", ".w")
+            // `vnsrl` is not in the above class
+            .replace("vnsrl.w", "vnsrl")
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:189]: oh my god

File: t1/src/decoder/Decoder.scala [hunk 5]
```diff
@@ -31,7 +33,186 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
...
+case class Op(tpe: String, funct6: String, funct3: String,
+              name: String, special: Option[SpecialAux], notLSU: Boolean) extends DecodePattern {
+  val funct3Map: Map[String, String] = Map(
+    "IV" -> "000",
+    "IX" -> "100",
+    "II" -> "011",
+    "MV" -> "010",
+    "MX" -> "110",
+    "FV" -> "001",
+    "FF" -> "101"
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:48]: OPCFG is missed, and please add documentation for these functions, give reference link and **align** to specification:
e.g.
```suggestion
    // Refer to [https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#101-vector-arithmetic-instruction-encoding].
    "OPIVV" -> "000",
    "OPFVV" -> "001",
    "OPMVV" -> "010",
    "OPIVI" -> "011",
    "OPIVX" -> "100",
    "OPFVF" -> "101",
    "OPMVX" -> "110",
    "OPCFG" -> "111",
```

File: t1/src/decoder/Decoder.scala [hunk 6]
```diff
@@ -31,7 +33,186 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
...
+  )
+
+  def bitPat: BitPat = if (notLSU) BitPat(
+    "b" +
+      // funct6
+      funct6 +
+      // TODO[0]: ? for vm
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:55]: finish this TODO?

File: t1/src/decoder/Decoder.scala [hunk 7]
```diff
@@ -31,7 +33,186 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
...
+    "IV" -> "000",
+    "IX" -> "100",
+    "II" -> "011",
+    "MV" -> "010",
+    "MX" -> "110",
+    "FV" -> "001",
+    "FF" -> "101"
+  )
+
+  def bitPat: BitPat = if (notLSU) BitPat(
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:51]: Please add doc to this function

File: t1/src/decoder/Decoder.scala [hunk 8]
```diff
@@ -31,7 +33,186 @@ trait BoolField extends BoolDecodeField[Op] with FieldName {
...
+  private val loadFPOpcode = "0000111"
+  private val storeFPOpcode = "0100111"
+
+  private val opivvFunct3 = "000"
+  private val opiviFunct3 = "011"
+  private val opivxFunct3 = "100"
+  private val opmvvFunct3 = "010"
+  private val opmvxFunct3 = "110"
+  private val opfvvFunct3 = "001"
+  private val opfvfFunct3 = "101"
```
Comment (sequencer) [inline t1/src/decoder/Decoder.scala:77]: Dup code?

Comment (Lucas-Wye) [reply t1/src/decoder/Decoder.scala]: already replaced with better name

**Review by sequencer — CHANGES_REQUESTED** [review]

**Review by sequencer — APPROVED** [review]

**Pull request merged**